src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Deployment/MicrosoftAccountDeploymentSource.cs

```diff
@@ -0,0 +1,33 @@
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using OrchardCore.Deployment;
+using OrchardCore.Microsoft.Authentication.Services;
+using OrchardCore.Microsoft.Authentication.Settings;
+
+namespace OrchardCore.Microsoft.Authentication.Deployment;
+
+public class MicrosoftAccountDeploymentSource : IDeploymentSource
+{
+    private readonly IMicrosoftAccountService _microsoftAccountService;
+
+    public MicrosoftAccountDeploymentSource(IMicrosoftAccountService microsoftAccountService)
+    {
+        _microsoftAccountService = microsoftAccountService;
+    }
+
+    public async Task ProcessDeploymentStepAsync(DeploymentStep step, DeploymentPlanResult result)
+    {
+        if (step is not MicrosoftAccountDeploymentStep)
+        {
+            return;
+        }
+
+        var settings = await _microsoftAccountService.GetSettingsAsync();
+
+        var obj = new JsonObject { ["name"] = nameof(MicrosoftAccountSettings) };
+
+        obj.Merge(JObject.FromObject(settings, JOptions.Default));
+
+        result.Steps.Add(obj);
```

Comment:
Did you actually try it? I don't think this will work. Also, there would be no way for the user to import that recipe.

I think it will have to be like this for it to be imported successfully.

```suggestion
        result.Steps.Add(new JsonObject
        {
            ["name"] = "Settings",
            [nameof(MicrosoftAccountSettings)] = JObject.FromObject(await _microsoftAccountService.GetSettingsAsync()),
        });
```